distributed/utils.py

```diff
@@ -1343,6 +1344,7 @@ def parse_ports(port):
     return ports
 
 
+@functools.lru_cache
 def is_coroutine_function(f):
     return asyncio.iscoroutinefunction(f) or gen.is_coroutine_function(f)
```

Comment:
Thoughts on just aliasing this?

```suggestion
is_coroutine_function = iscoroutinefunction
```

Reply:
I'm in favor. Looking through the history, it seems there was once some py3.5 compatibillity-related difference between the two, and there is no longer.